src/irmin-layers/stats.ml

```diff
@@ -15,113 +15,187 @@
  *)
 open! Import
 
-type t = {
-  mutable nb_freeze : int;
-  mutable copied_contents : int list;
-  mutable copied_nodes : int list;
-  mutable copied_commits : int list;
-  mutable copied_branches : int list;
-  mutable waiting_freeze : float list;
-  mutable completed_freeze : float list;
-  mutable skips : int;
-}
+(* This file avoids [option] type and clean functionnal paradigms in order to
+   lower the cpu footprint *)
+
+(** Ensure lists are not growing indefinitely by dropping elements. *)
+let limit_length_list = 10
 
-type i = {
+type freeze_profile = {
+  mutable idx : int;
+  mutable t0 : Mtime.t;
+  mutable t1 : Mtime.t;
+  mutable current_section : string;
+  mutable rev_timeline : (string * Mtime.t * float) list;
   mutable contents : int;
   mutable nodes : int;
   mutable commits : int;
   mutable branches : int;
   mutable adds : int;
+  mutable skips : int;
+  mutable copy_newies_loops : int;
+  mutable yields : int;
+  mutable outside_totlen : float;
+  mutable inside_totlen : float;
+  mutable outside_maxlen : string * float;
+  mutable inside_maxlen : string * float;
```

Comment:
it would be nice if we kept here all blocking portion that last more than 1s.

Reply:
Very good idea